nomad/event_sink_manager.go

```diff
@@ -402,148 +379,134 @@ func (m *ManagedSink) Run() error {
 				return
 			case <-m.stopCtx.Done():
 				return
-			case err := <-m.WatchCh():
+			case err := <-m.watch():
 				if err != nil {
 					return
 				}
 
-				// check if we should reload or just reset
-
-				reload := m.ResetWatchSet()
-
-				if !reload {
-					continue
-				}
-				// Cancel the subscription scoped context
-				m.cancelFn()
-
-				// wait until the reset was done
-				select {
-				case <-m.stopCtx.Done():
-					return
-				case <-m.doneReset:
-				case <-exitCh:
-				}
+				// WatchCh was triggered, reset the the WatchCh and potentially
+				// cancel the currentSinkCtx to reload changes
+				m.resetSink()
 			}
 		}
 	}()
 
-LOOP:
+START:
+	// Subscribe to event broker and establish SinkWriter
+	err := m.subscribe()
+	if err != nil {
+		return err
+	}
+
+	currentSinkCtx := m.sinkCtx
 	for {
-		events, err := m.Subscription.Next(m.sinkCtx)
+		events, err := m.subscription.Next(currentSinkCtx)
 		if err != nil {
 			// Shutting down, exit gracefully
 			if m.stopCtx.Err() != nil {
 				return m.stopCtx.Err()
 			}
 
-			// Reloadable error, reload and restart
-			if err == stream.ErrSubscriptionClosed || err == context.Canceled {
-				if err := m.Reload(); err != nil {
-					return err
-				}
-				goto LOOP
-			}
-			return err
+			m.logger.Info("received error from managed sink subscription, reloading sink", "error", err)
```

Comment:
I usually reserve `Info` for startup status only ("I'm listening on this port"). Operators who are shipping logs will filter and alert on these levels. Either it's just debugging info which should be at `Debug`, or its actionable in which case it should be at `Warn` or above. `Info` doesn't tell the operator what to do about it.

Reply:
:+1: makes sense, Debug should be fine here